src/lakefs_spec/spec.py

```diff
@@ -348,17 +348,22 @@ def info(self, path: str | os.PathLike[str], **kwargs: Any) -> dict[str, Any]:
             except ApiException as e:
                 raise translate_lakefs_error(e, rpath=path)
 
-        out = self.ls(path, detail=True, **kwargs)
+        out = self.ls(path, detail=True, recursive=True, **kwargs)
         if not out:
             raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), path)
 
         return {
             "name": path.rstrip("/"),
-            "size": sum(o.get("size", 0) for o in out),
+            "size": sum(o.get("size") or 0 for o in out),
```

Comment:
I prefer the old version - what was the problem with that? (This incurs another `None` to `int` comparison btw)

Reply:
When not fetching recursively (i.e., `delimiter="/"`), the API returns `size=None`, which would lead to an attempt to call `__add__` with `None` and `int` params in the old version.